janitor/functions.py

```diff
@@ -4245,3 +4245,133 @@ def fill_direction(
                 df.loc[:, column].ffill(limit=limit).bfill(limit=limit)
             )
     return df
+
+
+@pf.register_dataframe_method
+def groupby_topk(
+    df: pd.DataFrame,
+    groupby_column_name: Hashable,
+    sort_column_name: Hashable,
+    k: int,
+    sort_values_kwargs: Dict = None,
+) -> pd.DataFrame:
+    """
+    Return a dataframe that has the top `k` values grouped by
+    `groupby_column_name` and sorted by `sort_column_name`. Additional
+    parameters to the sorting (such as ascending=True) can be passed using
+    `sort_values_kwargs`. List of all sort_values() parameters can be found at
+    https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sort_values.html
```

Comment:
Let's get this adhering to [semantic line breaks][sembr]!

[sembr]: https://sembr.org/

Here's my suggested change:

```suggestion
    Return top `k` rows from a groupby of a set of columns.
    
    Returns a dataframe that has the top `k` values grouped by `groupby_column_name`
    and sorted by `sort_column_name`.
    Additional parameters to the sorting (such as ascending=True)
    can be passed using `sort_values_kwargs`.
    
    List of all sort_values() parameters can be found here_.
    
    .. _here: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sort_values.html
```

I did a few things in the suggestion:

1. Adhered to docstring conventions by making first line a single sentence.
2. Used ReStructured text URLs to reference the docs. (This is easier than copy/pasting the URL.) The documentation on how to insert a URL lives [here][rst_url].

[rst_url]: https://docutils.sourceforge.io/docs/user/rst/quickref.html#hyperlink-targets

Finally, you might need a `# noqa` on the docstring because the lines _might_ extend beyond the 79 character limit. You'll see sufficient examples in `functions.py` that have this `# noqa` piece.

Reply:
Hi, I did as requested and added # noqa E501. Hadn't heard of noqa and semantic line breaks before so this was very educational.